shopify-client/src/components/Header.vue

```diff
@@ -43,9 +46,18 @@ export default {
   methods:{
     submitSearch: function(){
       if(this.keyword){
+        let currRoute = this.$router.currentRoute.name;
         this.$router.push({name: 'ShopList', query:{keyword: this.keyword}});
+        if(currRoute === 'ShopList'){
+          this.$router.go(0);
```

Comment:
Why do you need to manually reload?

Reply:
if on the same page, it will only change the query param but will not reset the page state
so this reload only occurs if we are on the same page but there's a change in param